### PR TITLE
Modernize hydrophone

### PR DESCRIPTION
**Pull request by derrickburns**
This addresses some tech debt and uses some newer packages for dependency injection and initialization.

1. Remove references to hakken client library.
2. Upgrade to Go 1.14.
3. Remove travis references to S3 bucket for storing artifacts.
4. Implement dependency injection using fx.
5. Implement initialization using envconfig.
6. Update dependencies.
7. Add liveness probe.

Maybe we should stop vendoring dependencies?!

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Development
-FROM golang:1.12.7-alpine AS development
+FROM golang:1.14.7-alpine AS development
```
Comment (pazaan) [inline Dockerfile:2]: Should we just update to 1.14.9 while we're here?

File: api/setup.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
+package api
```
Comment (pazaan) [inline api/setup.go:1]: Rename file to indicate that it's for testing? Maybe `setup_test.go`?

File: hydrophone.go [hunk 3]
```diff
@@ -1,175 +1,172 @@
...
-	hakkenClient.Publish(&config.Service)
+	return &http.Client{Transport: tr}
+}
+
+func emailTemplateProvider() (models.Templates, error) {
+	emailTemplates, err := templates.New()
+	return emailTemplates, err
+}
 
-	signals := make(chan os.Signal, 40)
```
Comment (pazaan) [inline hydrophone.go:159]: Do we need to have this signal handler in the new code? Or is it managed as part of `fx`?

**Review by pazaan — COMMENTED** [review]
Some queries for my understanding before I press the OK button 😄

File: hydrophone.go (hunk 3 above)
Comment (derrickburns) [reply hydrophone.go:159]: It is in Fx

File: api/setup.go (hunk 2 above)
Comment (derrickburns) [reply api/setup.go]: Done

File: Dockerfile (hunk 1 above)
Comment (derrickburns) [reply Dockerfile]: Done

**Comment by derrickburns** [comment]
@pazaan Updated as requested.

**Review by jamesraby — APPROVED** [review]
Interesting to see fx conversion.

LGTM

**Comment by derrickburns** [comment]
@tjotala Tested in QA environment.

**Review by pazaan — APPROVED** [review]
LGTM 🚀

**Review by tjotala — APPROVED** [review]
LGTM :shipit:

**Pull request merged**